cmake/archdetect.cmake

```diff
@@ -0,0 +1,77 @@
+# archdetect.c -- Detect compiler architecture and set ARCH and BASEARCH
```

Comment:
Should be archdetect.cmake not archdetect.c

Reply:
Fixed